test/e2e/tests/test_trainingjob_debugger.py

```diff
@@ -0,0 +1,226 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Integration tests for the SageMaker TrainingJob API.
+"""
+
+import boto3
+import pytest
+import logging
+from typing import Dict
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+from e2e import (
+    service_marker,
+    create_sagemaker_resource,
+    wait_for_status,
+)
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.bootstrap_resources import get_bootstrap_resources
+from e2e.common.config import *
+from time import sleep
+
+RESOURCE_PLURAL = "trainingjobs"
+
+def _sagemaker_client():
+    return boto3.client("sagemaker")
+
+@pytest.fixture(scope="function")
+def xgboost_training_job_debugger():
+    resource_name = random_suffix_name("xgboost-trainingjob-debugger", 32)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["TRAINING_JOB_NAME"] = resource_name
+    reference, _, resource = create_sagemaker_resource(
+        resource_plural=RESOURCE_PLURAL,
+        resource_name=resource_name,
+        spec_file="xgboost_trainingjob_debugger",
+        replacements=replacements,
+    )
+
+    assert resource is not None
+    assert k8s.get_resource_arn(resource) is not None
+
+    yield (reference, resource) 
+
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted
+
+def get_sagemaker_training_job(training_job_name: str):
+    try:
+        training_job = _sagemaker_client().describe_training_job(
+            TrainingJobName=training_job_name
+        )
+        return training_job
+    except BaseException:
+        logging.error(
+            f"SageMaker could not find a training debugger job with the name {training_job_name}"
+        )
+        return None
+
+# TODO: Move to __init__.py
```

Comment:
nit: can we do it now?
not a blocker

Reply:
keeping as TODO for now since as discussed this may not be straightforward.

Reply:
I had a typo it should work

Reply:
I tried but still ran into issues using yield and parameterized fixtures. Plus I will need to change this across all tests. I'll create a separate PR but do not see value in spending too much time debugging this at the moment.